### ISSUE TITLE
Disney+ addon page view types & pagination (change request)

### ISSUE THREAD

**Issue by kevwag** [matthuisman/slyguy.addons#104]
Only four of the standard page views (Widelist / Wall / Infowall / Shift) are available when browsing content and you have to manually change the view type for every single page you visit. This gets particularly tedious since pagination cannot be turned off, meaning that for every click going to page2, page3, page4 etc, you have to change the view manually (unless you want the default view of course).
What I'm basically asking for breaks down into three parts:
1. a global view setting that will apply one view to all pages (or maybe two - one for shows and one for movies)
2. Inclusion of all the standard view types as per the default skin (I personally like using the "Poster" view, particularly when browsing for Movies).
3. Ability to turn off the annoying pagination or at least make it a more sensible number than 15 items per page.

Thanks in advance :)

**Comment by matthuisman** [comment]
can you advise of other addons that allow you to set the view like this?
Im not sure if its even possible

**Comment by kevwag** [comment]
Hi Matt,I can’t remember where I saw the option to turn off pagination but I’ve definitely seen it. I think it was something within the Official Kodi repo (sorry that’s a bit vague). If you don’t want to allow pagination to be turned off, could you at least make the number of items per page a bit more sensible? (The Netflix addon has 90 items per-page compared to the 15 you’ve set in Disney+). As for changing the browsing views, the Netflix addon by castagnait  has a “skin viewtypes” category in the addon settings but it’s a bit awkward as it requires you to know the numerical ids of the view-types supported by your skin. There’s a more user-friendly way to achieve the desired result but the only definite examples I can refer you to are in what Kodi call “banned addons”, namely “Venom”, “TheOath” and pretty much every other Exodus fork from the past few years. so I’ll also provide a description of how they implement the feature in these addons in case you’d prefer to avoid them: In these addons, the feature is not accessed from the settings menu, there is a “Tools” page in the addon and then the view setting is in there. When you click it, a dialogue asks you which media  type you want to set the view for (Movies / TV Shows / Seasons / Episodes), once you choose one it opens a standard navigation page as if you were browsing content but it only contains two thumbnails: one with the back arrow (or “..” in list views) to go back a page (effectively cancels the process), the other thumbnail is a save button – you select the view type in the usual way (say widelist for example) and then click the save thumbnail.  Now that view type will always be used for the type of content you chose in the first dialogue.I think they did it this way to make it work with all skins without the devs needing to know what views are available in every skin. Please feel free to get in touch if you need any more details etc.Regards, Kevin Wagener51-B Malvern Road,Bournemouth,Dorset,BH9 3AF (07527) ***@***.*** From: Matt HuismanSent: 17 December 2021 01:23To: matthuisman/slyguy.addonsCc: Kevin Wagener; AuthorSubject: Re: [matthuisman/slyguy.addons] Disney+ addon page view types & pagination (change request) (Issue #104) can you advise of other addons that allow you to set the view like this?Im not sure if its even possible—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by matthuisman** [comment]
For a start, pagination is upto Disney what they allow for api requests. 15 is what there own apps allow. I could always do multiple fetches so longer loading time but less pages. What would be nice is if Kodi had some sort of infinity scroll where it calls the addon when gets near bottom of list and fetches more results

**Comment by matthuisman** [comment]
As for the default view, I've seen how they do it. There is an Kodi api request to set view. So pretty simple. I think I'll make it a Slyguy common setting so you set it once and it applies to all slyguy add-ons.

**Comment by kevwag** [comment]
It kinda does do that in a way...It loads the thumbnails in the order that they appear in the list so in the case of the Netflix addon, sometimes I have to stop scrolling the list to allow the thumbnail loading to catch up. I know it’s not technically the same in terms of what’s actually going on in the background although it appears similar visually. As for the multiple fetches – that would probably be a good idea, it’s not like it really takes long. Three fetches would give you 45 shows which isn’t too bad.I’m sure I don’t need to tell you how frustrating it is trying to find a show again and you know it was buried somewhere like the tenth page into a category with no way of getting there other than to navigate through the preceding nine pages! Regards, Kevin Wagener51-B Malvern Road,Bournemouth,Dorset,BH9 3AF (07527) ***@***.*** From: Matt HuismanSent: 21 December 2021 17:57To: matthuisman/slyguy.addonsCc: Kevin Wagener; AuthorSubject: Re: [matthuisman/slyguy.addons] Disney+ addon page view types & pagination (change request) (Issue #104) For a start, pagination is upto Disney what they allow for api requests. 15 is what there own apps allow. I could always do multiple fetches so longer loading time but less pages. What would be nice is if Kodi had some sort of infinity scroll where it calls the addon when gets near bottom of list and fetches more results—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by kevwag** [comment]
Putting the view settings in the common settings module is an excellent idea.Can I just make you aware of something that might need looking at:Last week I thought you had actually done something regarding the view types because I was browsing Movies and the Poster view that I like was suddenly available. It turns out that what I’d missed is that some categories have more available views than others. When I browsed the documentaries category I was back to only having four available views, none of which were the one I wanted. I think this might be because the category isn’t one of the usual four view types (Movies / TV Shows / Seasons / Episodes), it’s a mixture of Movies AND TV Shows so if you’re going to implement the view settings on the basis of content type, you’ll need a fifth view type for “Mixed media types” I also noticed that while browsing, regardless of the view selected, no details or description of the currently highlighted show are displayed on the page. The data can be accessed from the “Information” option in the context menu but it would be cool if you could map the data to appear on the view page.  An indication of whether something is a movie or a TV series would also be really useful – the only way to tell at the moment is to open the info page and look to see if there’s a “Browse” button, indicating a series, or a “Play” button for movies. Regards, Kevin Wagener51-B Malvern Road,Bournemouth,Dorset,BH9 3AF (07527) ***@***.*** From: Matt HuismanSent: 21 December 2021 18:23To: matthuisman/slyguy.addonsCc: Kevin Wagener; AuthorSubject: Re: [matthuisman/slyguy.addons] Disney+ addon page view types & pagination (change request) (Issue #104) As for the default view, I've seen how they do it. There is an Kodi api request to set view. So pretty simple. I think I'll make it a Slyguy common setting so you set it once and it applies to all slyguy add-ons.—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by matthuisman** [comment]
you need to really fix your replies here. they are a huge mess. With a bunch of style crap etc.
Makes it nearly impossible to ready and just annoying as hell

**Comment by kevwag** [comment]
Sorry, I’m just replying in Windoze mail. I’ve deleted the signature file and reset the default font – is that any better?  From: Matt HuismanSent: 22 December 2021 00:13To: matthuisman/slyguy.addonsCc: Kevin Wagener; AuthorSubject: Re: [matthuisman/slyguy.addons] Disney+ addon page view types & pagination (change request) (Issue #104) you need to really fix your replies here. they are a huge mess. With a bunch of style crap etc.Makes it nearly impossible to ready and just annoying as hell—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by matthuisman** [comment]
nope, its still a mess. you may need to open github and reply directly on the site.

as for mixed content, thats treated the same as Menus.
So the Menu view will apply to them

**Comment by kevwag** [comment]
Ouch! I see what you mean now! the github mails I get from you state that you can reply directly to the email. They fail to mention that their system can't handle anything more than 1990's style plain-text emails!

Is there any way around that view setting for the mixed content at all? The documentaries in particular are difficult to browse when they're not actually being treated as media content.

**Comment by matthuisman** [comment]
i guess I could have another setting where you could tell it what to treat mixed content as (movie, tvshow, episode)

**Comment by matthuisman** [comment]
and did confirm, requesting any more than 15 items from their api causes them throw an error.
obviously so their DB isn't hammered.
So i'd need to do multi requests.

Again, be nice if Kodi add-ons had some sort of dynamic update system.
it would allow for things like
1) when a list item is showing - the add-on could fetch the full information for it and show it
2) when nearly the end of the list - addon fetches more results and adds them
(basically what the website / apps can do)

But anyway, for now I can add a setting for mutli-page fetch.
eg. 3 pages would be 45 results.
i can probably also request these in parallel threads for faster response

**Comment by matthuisman** [comment]
0.10.8 of Disney+ add-on just pushed.
This adds a new setting called "Pagination Multiplier".
Default for Disney+ is 3 (so it will get 3x pages at a time)
They are 15x items per page - so you should get 45 per page now.
You can set this new value to anything between 1 and 10.

Its also a nice generic approach, so I can easily add the setting to any of my add-ons

**Comment by kevwag** [comment]
Yes, just seen it, it's awesome! Thank you :)

As and when you tackle the view types, could you include the "poster" view type as an option for each view please? It seems a bit hit-and-miss. If you browse to Series>>Docuseries it's available as a view-type, along with the full complement of views available (8 in total), but if you browse to Hubs>>National Geographic>>Documentary Series, only four views are available and Poster isn't one of them.

Also, a quick question: 
In the "Look & Feel" section of the common settings module, What does the "Video view for all content" setting do? I've browsed with it off and then again with it turned on but I can't seem to work out what's different?

**Comment by matthuisman** [comment]
I can't control what Kodi gives you as view options. It's the content type
that controls that

That setting will set all content to be 'videos' instead of movies, tv
shows etc.
It means the view stays the same the whole way through the addon basically.
I guess makes it more like a file view than a library view.

Once I do the new view setting, it probably won't be needed anymore.

**Comment by kevwag** [comment]
Just after I posted that question I tried it again and realised that it removes half the available views from the menu's that had eight.
Why is Kodi making certain view-types unavailable to your addon based on content type when it doesn't inflict this restriction on other addons. I just did a quick check on 3 addons I use regularly; "Netflix", "Discovery+" and "Crackle" (from the official Kodi repo) and all three have the full complement of 8 views for all media types. Crackle in particular is very clearly delineated - the root menu only has two options, TV Shows and Movies, so there's no mixed content, it's decisively split into two sections. (In fact, the only addon I can think of that doesn't have all the views available is "Iplayer WWW" from the official Kodi repo and that addon barely works anyway. It's got a few big problems, which when I contacted the dev about, he had no interest in fixing because they affect things that he doesn't personally use).
Could you maybe replicate whatever trick those addons are using to de-restrict the view settings?

**Comment by kevwag** [comment]
...To add to that, the views are all available when I browse my local video library from the home screen. "TV Shows" and "Movies" menus have all the views available.

**Comment by matthuisman** [comment]
This is just temporary. Any mixed content will be using the video view. Before today if 70% of the content was one type, it would use that. But now needs to be 100% in preparation for the user adjustable views

**Comment by kevwag** [comment]
I think all the views were there for Movies>Comedy, yes. I can't check at the moment - I was zipping through and setting the views for my favourite categories as changing the common setting messed things up a bit. Anyway, I did that a bit too fast and earned myself a temporary ban on my IP! They didn't say how long "temporary" is, hopefully no more than an hour.

One list that I noticed doesn't have all the views is the watchlist. I'm guessing that's because it's not 100% one type as you just mentioned.

**Comment by matthuisman** [comment]
Who banned your IP? Disney?

**Comment by kevwag** [comment]
Yeah, the Kodi addon throws an api forbidden error so I opened the Windows app on my pc and it says:
"Due to an abnormally high number of requests, we have temporarily blocked traffic from your IP address. Please try again later (error code 91)"

**Comment by matthuisman** [comment]
Oh.. could be related to the new Pagination setting. Remember a 3 on that
does 3 requests per page

**Comment by kevwag** [comment]
No, it was definitely me going at it like a madman! The multiplier won't have helped matters but I don't want you to think that's what caused it. I got carried away.
Root cause = Me.

**Comment by kevwag** [comment]
Wow, so that was a full 24 hour lockout I got for doing that! 
Lesson learned - don't try opening every menu on the site in the space of 5 minutes lol

**Comment by Arndorferd** [comment]
> As for the default view, I've seen how they do it. There is an Kodi api request to set view. https://github.com/elgatito/plugin.video.elementum/blob/7745e20fd55d3ddf30efe0d2c63e696c5a86030c/resources/site-packages/elementum/navigation.py#L484
> 
> So pretty simple. I think I'll make it a Slyguy common setting so you set it once and it applies to all slyguy add-ons.
> 
> There will be View for Menus View for Shows View for Seasons View for Episodes View for Movies
> 
> However, your 2) in first post asks to include more views. Again, that's upto the skin. The addon simply tells it what content it's displaying. Eg. Movies. Therefore it should have all the views that Movies normally has

I looked in slyguy commons and just see a on/off setting for video view for all content. Im guessing thats not related and this feature is not implemented? If this portion isn't yet a feature I feel it would be a really good feature to implement especially if it is not too difficult as you make it sound.

**Comment by matthuisman** [comment]
closing this in favor of https://github.com/matthuisman/slyguy.addons/issues/700 as this issue has a bunch of off-topic.
